spec.bs

```diff
@@ -364,15 +458,29 @@ To get the <dfn>max batch size</dfn> for an [=/origin=] |issuer|, run the follow
 
 To <dfn>generate masked tokens</dfn> given [=key commitments=] |issuerKeys| and number |numTokens|, run the following steps:
 
-1. Let |maskOperation| be the Mask operation corresponding to |issuerKeys|["protocol_version"].
-1. Let |result| ([=tuple=] ([=byte sequence=], [=byte sequence=])) be the result of running |maskOperation| on |issuerKeys| and |numTokens|.
-1. If |result| is null, return an error.
-1. Return |result|.
-
-To <dfn>unmask tokens</dfn> given [=key commitments=] |issuerKeys|, byte string |pretokens|, and a bytestring |response|, run the following steps:
-
-1. Let |unmaskOperation| be the Unmask operation corresponding to |issuerKeys|["protocol_version"].
-1. Let |result| ([=list=] of byte strings) be the result of running |unmaskOperation| on |pretokens| and |response|.
+1. Let |issueRequest| be an empty [=IssueRequest=].
+1. Set |issueRequest|["count"] to |numTokens|.
+1. Let |blinds| be an empty [=list=].
+1. Repeat the following steps, |numTokens| times:
+    1. Let |input| be a random [=byte sequence=].
+    1. Let (|blind|, |blindedElement|) ([=tuple=] ([=byte sequence=], [=byte sequence=])) be the result of running the [=Blind=] function of the [[!VOPRF]] protocol on |input|, where |blindedElement| is encoded as a X9.62 uncompressed point.
+    1. [=list/Append=] the |blind| to |blinds|.
+    1. Append the second element of |blindedElement| to |issueRequest|["nonces"].
+1. Let |issueHeader| be the result of [=serialize protocol message|serializing protocol message=] |issueRequest|.
+1. Return [=tuple=] (|issueHeader|, |blinds|).
+
+To <dfn>unmask tokens</dfn> given [=key commitments=] |issuerKeys|, byte string |blinds|, and a [=byte sequence=] |response|, run the following steps:
+
+1. Let |input| be an empty [=byte sequence=].
+1. Let |evaluatedElements| be an empty [=list=].
+1. Let |blindedElements| be an empty [=list=].
+1. Let |issueResponse| be the result of [=deserialize protocol message|sdeerializing protocol message=] |response| as an [=IssueResponse=].
+1. [=list/For each=] |nonce| of |issueResponse|["signed"]:
+    1. [=list/Append=] |nonce|["blinded"] to |blindedElements|.
+    1. [=list/Append=] |nonce|["evaluated"] to |evaluatedElements|.
+1. Let |pkS| be |issuerKeys|["keys"][|issueResponse|["key_id"]]["Y"].
+1. Let |proof| be |issueResponse|["proof"].
+1. Let |result| ([=list=] of byte strings) be the result of running [=PSTFinalize=] on |input|, |blinds|, |evaluatedElements|, |blindedElements|, |pkS|, and |proof|.
```

Comment:
PSTFinalize uses a parameter 'G' (which I think identifies the [ciphersuite](https://www.ietf.org/archive/id/draft-irtf-cfrg-voprf-21.html#name-ciphersuites)?). How is that getting passed in?

Reply:
Added a line under VOPRF Methods since its defined for all VOPRF methods used by PST.